src/service/folder_service.rs

```diff
@@ -503,6 +533,66 @@ fn is_attempt_move_to_sub_child(
     }
 }
 
+/// Handles the removal of implicit tags from descendants when a folder is moved.
+///
+/// When a folder's parent changes, this function removes implicit tags from all
+/// descendant files and folders that originated from the old ancestor chain.
+///
+/// ## Parameters
+/// - `folder_id`: the id of the folder being moved
+/// - `original_ancestors`: list of ancestor folder ids from before the move
+///
+/// ## Returns
+/// - `Ok(())` if tags were successfully removed
+/// - `Err(UpdateFolderError)` if there was a database error
+fn handle_folder_move_for_tags(
```

Comment:
Add tests to make sure that 
- files and folders unaffected by the move don't have their tags removed